File: frontend/src/types/auditLog.ts
/**
 * Types pour le journal d'audit (entité `audit_log` côté backend).
 *
 * Voir `backend/app/schemas/audit_log.py` et `backend/app/models/audit_log.py`.
 */

export type AuditAction = 'create' | 'update' | 'delete';

export interface AuditLogEntry {
  id: number;
  occurredAt: string;
  userId: number | null;
  userEmail: string | null;
  action: AuditAction;
  entityType: string;
  entityId: string;
  beforeJson: Record<string, unknown> | null;
  afterJson: Record<string, unknown> | null;
  diffJson: Record<string, { before: unknown; after: unknown }> | null;
  ipAddress: string | null;
  userAgent: string | null;
  requestId: string | null;
}

export interface AuditLogListResponse {
  items: AuditLogEntry[];
  total: number;
  limit: number;
  offset: number;
}

export interface AuditLogFilters {
  entityType?: string;
  entityId?: string;
  userId?: number;
  action?: AuditAction;
  from?: string;
  to?: string;
  limit?: number;
  offset?: number;
}

File: backend/app/schemas/audit_log.py
"""Schémas Pydantic pour l'endpoint admin de lecture `audit_log`."""
from __future__ import annotations

from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class AuditLogRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    occurred_at: datetime
    user_id: int | None
    user_email: str | None
    action: str
    entity_type: str
    entity_id: str
    before_json: dict[str, Any] | None
    after_json: dict[str, Any] | None
    diff_json: dict[str, Any] | None
    ip_address: str | None
    user_agent: str | None
    request_id: str | None


class AuditLogListResponse(BaseModel):
    items: list[AuditLogRead]
    total: int
    limit: int
    offset: int
